src/less/expressive-loader/expressive-loader.less

```diff
@@ -36,141 +50,127 @@
-
-        &.expressive-loader__message--out {
-            animation: expressive-loader-message-fadeout
-                var(--expressive-loader-message-fadeout-duration)
-                cubic-bezier(0.3, 0, 0, 1) both;
-
-            @media (prefers-reduced-motion) {
-                animation: none;
-            }
-        }
-
-        &.expressive-loader__message--first {
-            @media (prefers-reduced-motion) {
-                animation: expressive-loader-message-fadein-rm
-                    var(--expressive-loader-message-fadein-duration)
-                    cubic-bezier(0.3, 0, 0, 1) both;
-            }
-        }
-    }
+// current message
+.expressive-loader__message[role="status"] {
+    opacity: 1;
+}
 
-    // next message
-    &[aria-hidden="true"] {
-        opacity: 0;
+// next message
+.expressive-loader__message[aria-hidden="true"] {
+    opacity: 0;
+}
 
-        &.expressive-loader__message--in {
-            animation: expressive-loader-message-fadein
-                var(--expressive-loader-message-fadein-duration)
-                cubic-bezier(0.3, 0, 0, 1) both;
-        }
+.expressive-loader__progress[role="progressbar"] {
+    border-radius: var(--expressive-loader-line-border-radius);
+    display: flex;
+    flex-shrink: 0;
+    gap: var(--expressive-loader-line-gap);
+    margin: 0 var(--spacing-200);
+    overflow: hidden;
+}
 
-        @media (prefers-reduced-motion) {
-            display: none;
+.expressive-loader__lines {
+    display: block;
+    font-size: 0;
+    height: var(--expressive-loader-line-height);
+    line-height: 0;
+    min-width: 100%;
+    white-space: nowrap;
+}
 
-            &.expressive-loader__message-in {
-                animation: none;
-            }
-        }
-    }
+.expressive-loader__line {
+    border-radius: var(--expressive-loader-line-border-radius);
+    display: inline-block;
+    height: var(--expressive-loader-line-height);
+    line-height: 0px;
+    transform-origin: 0;
 }
 
-.expressive-line-animation {
-    --height: 6px;
-    --line-gap: 4px;
-    --line-border-radius: 2px;
+.expressive-loader__line-1 {
+    background-color: var(--expressive-loader-color-1);
+}
 
-    .expressive-line-animation__line {
-        border-radius: var(--line-border-radius);
-        display: inline-block;
-        height: var(--height);
-        line-height: 0px;
+.expressive-loader__line-2 {
+    background-color: var(--expressive-loader-color-2);
+}
 
-        &--red {
-            background-color: var(--color-expressive-loader-red);
-        }
+.expressive-loader__line-3 {
+    background-color: var(--expressive-loader-color-3);
+}
 
-        &--blue {
-            background-color: var(--color-expressive-loader-blue);
-        }
+.expressive-loader__line-4 {
+    background-color: var(--expressive-loader-color-4);
+}
 
-        &--yellow {
-            background-color: var(--color-expressive-loader-yellow);
-        }
+@media (prefers-reduced-motion) {
```

Comment:
Moved all the specific reduced motion / non-reduced motion code here for clarity.